libs/plugins/flow-client/src/lib/core/state/flow/cases/update-on-drop.ts

```diff
@@ -0,0 +1,14 @@
+import { updateLinks } from '@flogo-web/lib-client/diagram';
+
+import { FlowState } from '../flow.state';
+import { getGraphName } from '../../utils';
+
+export function updateLinksOnDrop(state: FlowState, { handlerType, itemId, parentId }) {
+  const currentGraphName: string = getGraphName(handlerType);
+  const currentGraph = state[currentGraphName];
+
+  return {
+    ...state,
+    [currentGraphName]: updateLinks(currentGraph, itemId, parentId),
```

Comment:
I've mentioned it in the updateLinks file but there's an issues with the fact that `updateLinks` function is not a pure function (it mutates parameters) and it is being used in reducers